src/deployment/deploymentWebviewController.ts

```diff
@@ -61,6 +62,10 @@ export class DeploymentWebviewController extends FormWebviewController<
                 ),
             },
         });
+        // If an initial connection group was provided, try to pre-populate the form state
+        if (initialConnectionGroup && initialConnectionGroup.id) {
+            (this.state.formState as any).groupId = initialConnectionGroup.id;

```

Comment:
This should go inside the `initialize()` function because it's initialization logic.